clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1513,7 +1513,8 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     const llvm::Triple::SubArchType DeviceSubArch = DeviceTriple.getSubArch();
     if (DeviceTriple.isNVPTX() || DeviceTriple.isAMDGPU() ||
         (DeviceTriple.isSPIR() &&
-         DeviceSubArch != llvm::Triple::SPIRSubArch_fpga))
+         DeviceSubArch != llvm::Triple::SPIRSubArch_fpga) ||
```

Comment:
Can you add a FE test? I assume there is already a FE test checking the existence of this macro for other targets. You can just add a RUN there.

Reply:
Added it, thank you